File: asml/LC566ReshapeTheMatrix.cpp
class Solution {
public:
    vector<vector<int>> matrixReshape(vector<vector<int>>& nums, int r, int c) {
        int m = nums.size();
        int n = nums[0].size();
        int product = m*n;
        if (r*c != product) return nums;
        vector<vector<int>> answer(r, vector<int>(c, 0));
        for (int i = 0; i < product; i++) {
            answer[i/c][i%c] = nums[i/n][i%n];
        }
        return answer;
    }
};

File: asml/LC566ReshapeTheMatrix.py
class Solution(object):
    def matrixReshape(self, nums, r, c):
        m = len(nums)
        n = len(nums[0])
        if m*n != r*c:
            return nums
        answer = []
        candidate = reduce(lambda x, y: x+y, nums)
        k = 0
        for i in range(r):
            temp = []
            for j in range(c):
                temp.append(candidate[k])
                k += 1
            answer.append(temp)
        return answer
        
